cli/spcli/sectors.go

```diff
@@ -876,84 +877,96 @@ func SectorsExtendCmd(getActorAddress ActorAddressGetter) *cli.Command {
-										(currEpoch <= sectorInfo.Expiration-builtin.EndOfLifeClaimDropPeriod) {
-										fmt.Printf("skipping sector %d because claim %d (client f0%s, piece %s) does not live long enough \n", sectorNumber, claimId, claim.Client, claim.Data)
-										cannotExtendSector = true
-										break
+									claimExpiration := claim.TermStart + claim.TermMax
+									// can be maintained in the extended sector
+									if claimExpiration > newExp {
+										claimIdsToMaintain = append(claimIdsToMaintain, claimId)
+									} else {
+										sectorInfo, ok := activeSectorsInfo[sectorNumber]
+										if !ok {
+											return xerrors.Errorf("failed to find sector in active sector set: %w", err)
+										}
+										if !cctx.Bool("drop-claims") ||
+											// FIP-0045 requires the claim minimum duration to have passed
+											currEpoch <= (claim.TermStart+claim.TermMin) ||
+											// FIP-0045 requires the sector to be in its last 30 days of life
+											(currEpoch <= sectorInfo.Expiration-builtin.EndOfLifeClaimDropPeriod) {
+											fmt.Printf("skipping sector %d because claim %d (client f0%s, piece %s) does not live long enough \n", sectorNumber, claimId, claim.Client, claim.Data)
+											cannotExtendSector = true
+											break
+										}
+
+										claimIdsToDrop = append(claimIdsToDrop, claimId)
 									}
 
-									claimIdsToDrop = append(claimIdsToDrop, claimId)
+									numbersToExtend = append(numbersToExtend, sectorNumber)
+								}
+								if cannotExtendSector {
+									continue
 								}
 
-								numbersToExtend = append(numbersToExtend, sectorNumber)
-							}
-							if cannotExtendSector {
-								continue
+								if len(claimIdsToMaintain)+len(claimIdsToDrop) != 0 {
+									sectorsWithClaims = append(sectorsWithClaims, miner.SectorClaim{
+										SectorNumber:   sectorNumber,
+										MaintainClaims: claimIdsToMaintain,
+										DropClaims:     claimIdsToDrop,
+									})
+								}
 							}
+						}
 
-							if len(claimIdsToMaintain)+len(claimIdsToDrop) != 0 {
-								sectorsWithClaims = append(sectorsWithClaims, miner.SectorClaim{
-									SectorNumber:   sectorNumber,
-									MaintainClaims: claimIdsToMaintain,
-									DropClaims:     claimIdsToDrop,
-								})
-							}
+						sectorsWithoutClaimsCount, err := sectorsWithoutClaimsToExtend.Count()
+						if err != nil {
+							return xerrors.Errorf("failed to count cc sectors: %w", err)
 						}
-					}
 
-					sectorsWithoutClaimsCount, err := sectorsWithoutClaimsToExtend.Count()
-					if err != nil {
-						return xerrors.Errorf("failed to count cc sectors: %w", err)
-					}
+						sectorsInDecl := int(sectorsWithoutClaimsCount) + len(sectorsWithClaims)
+						scount += sectorsInDecl
 
-					sectorsInDecl := int(sectorsWithoutClaimsCount) + len(sectorsWithClaims)
-					scount += sectorsInDecl
+						if scount > addrSectors || len(p.Extensions) >= declMax {
```

Comment:
I don't see a point for this check. If we are already batching and checking that batch size is smaller than policy max then what's the point of this check?

Reply:
I understand what you mean, this part is the original logic, so I didn’t change it.
The first half of this judgment is redundant, and the second half of the inspection may still be useful. I had no one to communicate with at the time, so I chose not to adjust it in a conservative strategy.

Reply:
Should I delete the redundant first half?

Reply:
Yes

Reply:
No, I realized, there is nothing wrong here, it is completely wrong to remove the first half.

Reply:
The function here is to determine whether a new parameter should be opened. The function of the previous batching is to reduce the input batch size.

Reply:
It's raining, 2300ml at a time
My barrel only has 500ml, and every time it rains 2300ml. My barrel is full, and there is still 1800ml (2300-500) left, and my other barrel cannot hold it.
So I divided the 2300ml each time into 500ml each time, so that my barrel can work normally (at this time, I also need to check whether the barrel is full)